gammapy/estimators/map/ts.py

```diff
@@ -745,6 +812,88 @@ def estimate_errn_errp(self, dataset, result):
         )
         return {"norm_errn": flux_errn, "norm_errp": flux_errp}
 
+    def sample_norm(self, result):
+        """Sample different scales to compute profile."""
+
+        norm_err = result["norm_err"]
+        norm = result["norm"]
+        if ~np.isfinite(norm_err) or norm_err == 0:
+            norm_err = 0.1
+        if ~np.isfinite(norm) or norm == 0:
+            norm = 1.0
+        sparse_norms = np.concatenate(
+            (
+                norm + np.linspace(-2.5, 2.5, 51) * norm_err,
+                norm + np.linspace(-10, 10, 21) * norm_err,
+                np.abs(norm) * np.linspace(-10, 10, 21),
+                np.linspace(-10, 10, 21),
+                np.linspace(self.norm.scan_values[0], self.norm.scan_values[-1], 2),
+            )
+        )
+        sparse_norms = np.unique(sparse_norms)
+        if len(sparse_norms) != 109:
+            rand_norms = 20 * np.random.rand(109 - len(sparse_norms)) - 10
+            sparse_norms = np.concatenate((sparse_norms, rand_norms))
+        return np.sort(sparse_norms)
+
+    def estimate_scan(self, dataset, result):
+        """Compute likelihood profile on a fixed norm range.
+
+        Parameters
+        ----------
+        dataset : `SimpleMapDataset`
+            Simple map dataset.
+
+        Returns
+        -------
+        result : dict
+            Result dictionary including 'stat_scan'.
+        """
+
+        sparse_norms = self.sample_norm(result)
+
+        scale = sparse_norms[None, :]
+        model = dataset.model.ravel()[:, None]
+        background = dataset.background.ravel()[:, None]
+        counts = dataset.counts.ravel()[:, None]
+        stat_scan = cash(counts, model * scale + background)
+
+        stat_scan_local = stat_scan.sum(axis=0) - result["stat_null"]
+
+        spline = InterpolatedUnivariateSpline(
+            sparse_norms, stat_scan_local, k=1, ext="raise", check_finite=True
+        )
+        stat_scan = spline(self.norm.scan_values)
+
+        ts = -stat_scan.min()
+        ind = stat_scan.argmin()
+        norm = self.norm.scan_values[ind]
+
+        maskp = self.norm.scan_values > norm
+        invalid_value = 999
+        stat_diff = stat_scan - stat_scan.min()
+        ind = np.abs(stat_diff + invalid_value * maskp - self.n_sigma**2).argmin()
```

Comment:
why not taking the argmin after applying the mask?

Reply:
done